### PR TITLE
fix response is not of given type

### LINKED ISSUE #30: Failed to unlock: The response is not of given type
As of version 0.6.0, when unlocking the key, I get the following error as a VSCode notification.

> Failed to unlock: The response is not of given type

This is what is displayed in the Output panel:

```
Try to unlock current key: <GPG>
Assuan data receive: OK Pleased to meet you, process 13156
Assuan data send: OPTION pinentry-mode loopback
Assuan data receive: OK
Assuan data send: SIGKEY <SIGKEY>
Assuan data receive: OK
Assuan data send: SETHASH --hash=sha1 <sha1>
Assuan data receive: OK
Assuan data send: PKSIGN
Assuan data receive: S INQUIRE_MAXLEN 255
Assuan data receive: INQUIRE PASSPHRASE
Assuan data send: D <passphrase>
Assuan data send: END
Assuan data receive: ERR 67108875 Bad passphrase <GPG Agent>
```

If I revert to 0.5.0 it unlocks without issue.

My passphrase uses special characters like `|#%*`.

### PR DESCRIPTION
**Pull request by zanix**
- Fix error when a percent character is in the passphrase

Typescript/Javascript is not my strongest language so I'm not sure if this is the best way to fix #30, but it works.

### CODE REVIEW

**Comment by wdhongtw** [comment]
LGTM. :D

**Review by wdhongtw — APPROVED** [review]
LGTM. :D

File: src/extension.ts [hunk 1]
```diff
@@ -295,6 +295,9 @@ class KeyStatusManager {
         }
 
         this.#logger.info(`Try to unlock current key: ${theKey.fingerprint}`);
+        // Escape percent character.
+        // https://www.gnupg.org/documentation/manuals/assuan/Client-requests.html#Client-requests
+        passphrase.replace(/%/g,'%25');
```
Comment (zanix) [inline src/extension.ts:300]: I'm so sorry, I just realized I messed this up. The code I submitted doesn't exactly match
This needs to be
```javascript
passphrase = passphrase.replace(/%/g,'%25');
```

**Comment by zanix** [comment]
I added another PR here #35

**Pull request merged**